rosidl_generator_py/resource/_msg.py.em

```diff
@@ -135,45 +136,44 @@ class @(message.structure.type.name)(metaclass=Metaclass_@(message.structure.typ
 @[end for]@
     ]
 
-    _fields_and_field_types = {
+    SLOT_TYPES = (
 @[for member in message.structure.members]@
 @{
 type_ = member.type
 if isinstance(type_, NestedType):
     type_ = type_.basetype
 }@
-        '@(member.name)': '@
-@# the prefix for nested types
-@[  if isinstance(member.type, Sequence)]@
-sequence<@
+        @
+@[  if isinstance(member.type, BoundedSequence)]@
+rosidl_parser.definition.BoundedSequence(@
```

Comment:
Instead of manually writing the name of each type I would suggest using `str(type(member.type))`.